src/pages/theming/css-shadow-parts.md

```diff
@@ -0,0 +1,153 @@
+- It contains children elements. For example, `ion-card-header` is a Shadow component, but all styles are applied to the host element. Since it has no child elements, there’s no need for parts.
+- The children elements are not structural. In certain components, including `ion-title`, the child element is a structural element used to position the inner elements. We do not recommend customizing structural elements as this can have unexpected results.
+
+> We welcome recommendations for additional parts. Please create a <a href="https://github.com/ionic-team/ionic-framework/issues/new?assignees=&labels=&template=feature_request.md&title=feat%3A+" target="_blank" rel="noopener noreferrer">new GitHub issue</a> with as much information as possible when requesting a part.
+
+## Known Limitations
+
+### Browser Support
+
+CSS Shadow Parts are supported in the recent versions of all of the major browsers. However, some of the older versions do not support shadow parts. Verify the <a href="https://caniuse.com/#feat=mdn-css_selectors_part" target="_blank" rel="noopener noreferrer">browser support</a> meets the requirements before implementing parts in an app. If browser support for older versions is required, we recommend continuing to use <a href="/docs/theming/css-variables" target="_blank" rel="noopener noreferrer">CSS Variables</a> for styling.
+
+### Vendor Prefixed Pseudo-Elements
+
+<a href="https://developer.mozilla.org/en-US/docs/Glossary/Vendor_Prefix" target="_blank" rel="noopener noreferrer">Vendor prefixed</a> pseudo-elements are not supported at this time. An example of this would be any of the `::-webkit-scrollbar` pseudo-elements:
+
+```css
+/* Does NOT work */
+my-component::part(scroll)::-webkit-scrollbar {
+  background: green;
+}
+```
+
+See <a href="https://github.com/w3c/csswg-drafts/issues/4530" target="_blank" rel="noopener noreferrer">this issue on GitHub</a> for more information.
+
+
+### Structural Pseudo-Classes
+
+Most pseudo-classes are supported with parts, however, <a href="https://www.w3.org/TR/selectors-4/#structural-pseudos" target="_blank" rel="noopener noreferrer">structural pseudo-classes</a> are not. An example of structural pseudo-classes that do not work is below.
+
+```css
+/* Does NOT work */
+my-component::part(container):first-child {
+  background: green;
+}
+
+/* Does NOT work */
+my-component::part(container):last-child {
+  background: green;
+}
+```
+
+### Chaining Parts
+
+The `::part()` pseudo-element can not match additional `::part()`s.
+
+For example, `my-component::part(button)::part(label)` does not match anything. This is because doing so would expose more structural information than is intended.
+
+If the `<my-component>`’s internal button uses something like `part="label => button-label"` to forward the button’s internal parts up into the panel’s own part element map, then a selector like `my-component::part(button-label)` would select just the one button’s label, ignoring any other labels.
```

Comment:
I haven't seen this syntax before `label => button-label`. Is there an MDN doc we can link to where people can read more?

Reply:
I got it from Example 4 here: https://www.w3.org/TR/css-shadow-parts-1/#part

Reply:
Ok that should be fine. I guess it just doesn't make a lot of sense to me - the spec doesn't give much information as to the `=>` syntax.

Reply:
I agree it's confusing. It's like the following:

```html
<my-component>
  #shadow-root
    <my-button part="button">
      #shadow-root
      <div part="label"></div>
    </my-button>
</my-component>
```

So this wouldn't work:

```css
my-component::part(button)::part(label) {
```

but if the button somehow forwards the part then it would work like:

```css
my-component::part(button-label)
```

I guess it would look like this:

```html
<my-component>
  #shadow-root
    <my-button part="button label => button-label">
      #shadow-root
      <div part="label"></div>
    </my-button>
</my-component>
```

I need to test this out though and then maybe I could add more usage. I don't think it applies to Ionic components right now.

Reply:
Ohh I see ok that makes sense. Yeah I don't think we use this in the framework anywhere?

Reply:
Not right now. This could maybe end up being a problem for things like `ion-infinite-scroll-content`, for example, if we allowed them to target parts on the spinner.